typings/index.d.ts

```diff
@@ -1487,6 +1517,45 @@ export class MessageButton extends BaseMessageComponent {
   private static resolveStyle(style: MessageButtonStyleResolvable): MessageButtonStyle;
 }
 
+export class ModalInputText extends BaseMessageComponent {
+  // TODO: Add API Types
+  public constructor(data?: ModalInputText | ModalInputTextOptions);
+  public style: ModalInputTextStyle;
+  public customId: string;
+  public label: string;
+  public placeholder: string | null;
+  public minLength: number | null;
+  public maxLength: number | null;
+  public setCustomId(customId: string): this;
+  public setLabel(label: string): this;
+  public setStyle(style: ModalInputTextStyleResolvable): this;
+  public setPlaceholder(placeholder: string): this;
+  public setMinLength(minLength: number): this;
+  public setMaxLength(maxLength: number): this;
+  // TODO: Add API Types
+  public toJSON(): unknown;
+  private static resolveStyle(style: ModalInputTextStyleResolvable): ModalInputTextStyle;
+}
+
+export class Modal {
+  // TODO: Add API types.
+  public constructor(data?: ModalOptions | ModalOptions);
+  public title: string;
+  public customId: string;
+  public components: MessageActionRowComponentResolvable[];
```

Comment:
```suggestion
  public components: MessageActionRow[];
```
A modal's components exists only of message action rows with one component each (as of now only INPUT_TEXT).

Reply:
Going to use a more strongly typed version of this.

Reply:
Actually scratch that